### PR TITLE
fix: Ensure that `testthat::skip()` is only called in RSQLite's own tests, never in the tests for other packages

### LINKED ISSUE #515: Unexpected message and test skipping when using `extended_types = TRUE`
I'm using RSQLite as the backend data store/access for a package designed to retrieve and access some fish data. When running tests, which involve creating the database using RSQLite as part of the setup, I saw some strange output (` {.fn {top_level_fun}} needs the {.pkg {.val  {pkgs_not_installed}}} package{?s}.` and noticed that most of my tests were being skipped. You can recreate the behavior by doing the below - my apologies for the complexity I'm not sure if I could recreate it outside of the packaging context.

I believe that the cryptic message itself is generated [here](https://github.com/r-dbi/RSQLite/blob/eb200266eaaa6a1fa83ec5e73770c8b9c7f01914/R/import-standalone-check_suggested.R#L63), and the actual problem is that I didn't have the `hms` package installed, which [this line](https://github.com/r-dbi/RSQLite/blob/35855bbc5b19f2b907108e43684f535e689ac24c/R/dbConnect_SQLiteDriver.R#L109) checks for. Additionally, I think that `dbConnect` is not a package anymore? 

Finally, I also question whether it is appropriate to force users to skip tests - had we not had other tests failing downstream because of the earlier skipped setup tasks, we might not have realized that all of our tests were being skipped on GHA.

In the end, I was able to resolve the issue by adding `hms` to Suggests. However, this behavior seems not ideal and it took me forever to figure out what the heck was happening, so I thought I'd bring it to your attention.

### Reprex

```r
# from outside a project
remove.packages("hms")
usethis::create_package("testRSQLbug")
# switch to testRSQLbug project
usethis::use_package("RSQLite")
usethis::use_package("DBI")
usethis::use_testthat()
usethis::use_test("test_connect")
```

```r
test_that("test connect", {
    mydb <- DBI::dbConnect(RSQLite::SQLite(), "my-db.sqlite", extended_types = TRUE)
    expect_s4_class(mydb, "SQLiteConnection")
    DBI::dbDisconnect(mydb)
})
```

output from testing

```
==> devtools::test()

ℹ Testing testRSQbug
✔ | F W  S  OK | Context
✔ |      1   0 | test_package_loading                        

══ Results ══════════════════════════════════════════════════
── Skipped tests (1) ────────────────────────────────────────
• {.fn {top_level_fun}} needs the {.pkg {.val
  {pkgs_not_installed}}} package{?s}. (1):
  test-test_package_loading.R:4:5

[ FAIL 0 | WARN 0 | SKIP 1 | PASS 0 ]
```

### PR DESCRIPTION
**Pull request by jeanetteclark**
Fixes #515 

- added `setup.R` to set package_name option and unset it after test are run
- removed `dbConnect` from list of packages to check (AFAICT it doesn't exist anymore)
- minor doc update that cropped up after running R CMD check

remaining tasks for #515 

- add an `&` to the [conditional](https://github.com/cynkra/dm/blob/784f778692494d16f7dccd7b53b57d6020defa9b/R/standalone-check_suggested.R#L60) with `getOptions("package_name") == "RSQLite"`
- wrap [malformed message](https://github.com/cynkra/dm/blob/784f778692494d16f7dccd7b53b57d6020defa9b/R/standalone-check_suggested.R#L62) in `glue`

### CODE REVIEW

File: NAMESPACE [hunk 1]
```diff
@@ -84,6 +84,7 @@ exportMethods(sqlData)
 import(DBI)
 import(methods)
 import(rlang)
+importFrom(DBI,Id)
```
Comment (krlmlr) [inline NAMESPACE:87]: Interesting, where does this come from?

File: R/dbConnect_SQLiteDriver.R [hunk 2]
```diff
@@ -106,7 +106,7 @@ dbConnect_SQLiteDriver <- function(drv, dbname = "", ..., loadable.extensions =
 
   extended_types <- isTRUE(extended_types)
   if (extended_types) {
-    check_suggested("hms", "dbConnect")
+    check_suggested("hms")
```
Comment (krlmlr) [inline R/dbConnect_SQLiteDriver.R:109]: `"dbConnect"` is used as the `top_level_fun` argument, it's not a package. I think the existing code is fine?

File: tests/testthat/setup.R [hunk 3]
```diff
@@ -0,0 +1,3 @@
+options(package_name = "RSQLite")
+
+withr::defer(options(package_name = NULL), teardown_env())
```
Comment (krlmlr) [inline tests/testthat/setup.R:3]: A bit more context, perhaps:

```suggestion
rlang::local_options(test_package_name = "RSQLite", .frame = teardown_env())
```

**Review by krlmlr — COMMENTED** [review]
Thanks, good start. Would you like to add the rest of the implementation to this PR?

For the next steps: Inside `if ()`,  `&&` is always better than `&` .

**Comment by jeanetteclark** [comment]
>  Would you like to add the rest of the implementation to this PR?

As discussed in #515, the rest of the changes have to go into `dm`. You mentioned that you would make the changes over there but I can also make the PR if you prefer, I think its just one line change. Let me know

File: NAMESPACE (hunk 1 above)
Comment (krlmlr) [reply NAMESPACE:87]: Note to self: this should have happened automatically, but hasn't due to "The decor package(s) are required for this functionality": https://github.com/r-dbi/RSQLite/actions/runs/9949313875/job/27485358644#step:11:75 .

**Review by krlmlr — COMMENTED** [review]
Thanks. Happy to review everything we want to do here as a single PR here, I can easily move to dm then.

File: R/import-standalone-check_suggested.R [hunk 4]
```diff
@@ -58,9 +58,9 @@ check_suggested <- function(packages, top_level_fun, use = TRUE) {
 
   # Skip if some packages are not installed when testing
   # And say which package was not installed.
-  if (identical(Sys.getenv("TESTTHAT"), "true")) {
+  if (identical(Sys.getenv("TESTTHAT"), "true") && identical(getOption("test_package_name"), "RSQLite")) {
```
Comment (krlmlr) [inline R/import-standalone-check_suggested.R:61]: How can we dynamically query the package name? Perhaps with `environmentName(parent.env(environment()))` ?

File: R/import-standalone-check_suggested.R [hunk 5]
```diff
@@ -58,9 +58,9 @@ check_suggested <- function(packages, top_level_fun, use = TRUE) {
...
     pkgs_not_installed <- packages[!installed]
-    message <- "{.fn {top_level_fun}} needs the {.pkg {.val {pkgs_not_installed}}} package{?s}."
+    message <- cli::cli_li("{.fn {top_level_fun}} needs the {.pkg {.val {pkgs_not_installed}}} package{?s}.")
```
Comment (krlmlr) [inline R/import-standalone-check_suggested.R:63]: Is this better than `glue::glue()` ?

**Review by krlmlr — COMMENTED** [review]
Thanks! Can you please also update the header of the standalone file to include the changes and the new dependency?

File: R/import-standalone-check_suggested.R (hunk 5 above)
Comment (jeanetteclark) [reply R/import-standalone-check_suggested.R:63]: Glue doesn't work with the string as written (syntax errors), but `cli::cli_li` does so I assumed that's what was meant here

File: R/import-standalone-check_suggested.R (hunk 4 above)
Comment (jeanetteclark) [reply R/import-standalone-check_suggested.R]: seems like `getPackageName()` does the trick (didn't know this existed until now). what you listed above returns an empty string within tests but maybe it just needs tweaking

**Comment by jeanetteclark** [comment]
not sure why that test is failing, seems unrelated? here's the output

File: R/import-standalone-check_suggested.R [hunk 6]
```diff
@@ -58,9 +58,9 @@ check_suggested <- function(packages, top_level_fun, use = TRUE) {
 
   # Skip if some packages are not installed when testing
   # And say which package was not installed.
-  if (identical(Sys.getenv("TESTTHAT"), "true")) {
+  if (identical(Sys.getenv("TESTTHAT"), "true") && identical(getOption("test_package_name"), getPackageName())) {
```
Comment (krlmlr) [inline R/import-standalone-check_suggested.R:61]: This is nice, but would have to be `methods::getPackageName()` (why is this not flagged?), and that function does too much too.

Essentially, `methods::getPackageName()` calls `environmentName(topenv())`, that looks better to me than my initial proposal.

**Review by krlmlr — COMMENTED** [review]
Thanks for the updates and for your patience. Should we do one more round?

There's a "Changelog" section in the header that could use a bullet.

This particular test is flaky, I need to review it at some point.

**Comment by jeanetteclark** [comment]
No worries. I'll work that in at some point today.

One thing to note - I removed the hms package and ran the tests locally, but a bunch still failed. I'm not sure how far this particular bit of code is supposed to go in terms of skipping tests, but all the failures were from `DBItest`.

**Comment by krlmlr** [comment]
Thanks for pushing through!

I don't understand why removing hms fails the tests because this package is really only used with specific `dbConnect()` arguments, AFAICT. And there's a successful run on CI/CD without hms: https://github.com/r-dbi/RSQLite/actions/runs/9979145876/job/27577869756 .

**Pull request merged**